build.assets/tooling/cmd/release-notes/release_notes.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	_ "embed"
+	"html/template"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/gravitational/trace"
+)
+
+var (
+	versionMatcher      = regexp.MustCompile(`\d+\.\d+\.\d+`)
+	doubleHeaderMatcher = regexp.MustCompile(`^##\s+`)
+)
+
+//go:embed template/release-notes.md.tmpl
+var tmpl string
+
+type tmplInfo struct {
+	Version     string
+	Description string
+}
+
+var (
+	releaseNotesTemplate = template.Must(template.New("release notes").Parse(tmpl))
+)
+
+type releaseNotesGenerator struct {
+	// releaseVersion is the version for the release.
+	// This will be compared against the version present in the changelog.
+	releaseVersion string
+}
+
+func (r *releaseNotesGenerator) generateReleaseNotes(md *os.File) (string, error) {
+	desc, err := r.parseMD(md)
+	if err != nil {
+		return "", err
+	}
+
+	info := tmplInfo{
+		Version:     r.releaseVersion,
+		Description: desc,
+	}
+	var buff bytes.Buffer
+	if err := releaseNotesTemplate.Execute(&buff, info); err != nil {
+		return "", trace.Wrap(err)
+	}
+	return buff.String(), nil
+}
+
+// parseMD is a simple implementation of a parser to extract the description from a changelog.
+// Will scan for the first double header and pull the version from that.
+// Will pull all information between the first and second double header for the description.
+func (r *releaseNotesGenerator) parseMD(md *os.File) (string, error) {
+	sc := bufio.NewScanner(md)
+	var buff bytes.Buffer
+
+	// Skip until first double header which should be version
+	for sc.Scan() {
+		if isDoubleHeader(sc.Text()) {
+			break
+		}
+	}
+
+	if err := r.checkVersion(sc.Text()); err != nil {
```

Comment:
I did the regex as a way to just ignore any potential trailing spaces in the header.

e.g. with extra space which is valid Markdown and will render without the extra spaces.
```
##  16.0.4
```
This would error with a simple split. I don't think this is likely to happen since the changelog is auto generated but I wouldn't expect to fail because of that.

It's originally why I did the double header regex to account for the trailing spaces but then I thought maybe it would be better to just match the version. Regex for semver matching is heinous though so I do think it would be better to do string matching.